File: test/language/templates/cpp98/InstantiateViaImportTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_via_import/InstantiateViaImport.h"

namespace templates
{
namespace instantiate_via_import
{

TEST(InstantiateViaImportTest, readWrite)
{
    InstantiateViaImport instantiateViaImport;
    pkg::U32 u32;
    u32.setValue(13);
    instantiateViaImport.setTest32(u32);
    pkg::Test_string str;
    str.setValue("test");
    instantiateViaImport.setTestStr(str);

    zserio::BitStreamWriter writer;
    instantiateViaImport.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateViaImport readInstantiateViaImport(reader);

    ASSERT_TRUE(instantiateViaImport == readInstantiateViaImport);
}

} // namespace instantiate_via_import
} // namespace templates

File: test/language/templates/java/templates/InstantiateViaImportTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_via_import.InstantiateViaImport;
import templates.instantiate_via_import.pkg.U32;
import templates.instantiate_via_import.pkg.Test_string;

public class InstantiateViaImportTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateViaImport instantiateViaImport = new InstantiateViaImport();
        instantiateViaImport.setTest32(new U32(13));
        instantiateViaImport.setTestStr(new Test_string("test"));

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        instantiateViaImport.write(writer);

        final BitStreamReader reader = new ByteArrayBitStreamReader(
                writer.toByteArray(), writer.getBitPosition());
        final InstantiateViaImport readInstantiateViaImport = new InstantiateViaImport(reader);
        assertTrue(instantiateViaImport.equals(readInstantiateViaImport));
    }
}
